### PR TITLE
Test dacify and reduce loops through waveforms.

### PR DESCRIPTION
**Pull request by DanielSank**
@JulianSKelly

### CODE REVIEW

File: fpgalib/dac.py [hunk 1]
```diff
@@ -85,41 +85,43 @@
...
+
+def dacify(wave_a, wave_b, trigger_idx=None):
+    """Construct SRAM sequence for a list of waveforms (DACify waveforms).
 
     Warning: Lots of hardware details exposed here.
 
     Args:
-        wave_a (np.ndarray): Waveform for DAC A normalized to ???
-        wave_b (np.ndarray): Waveform for DAC B normalized to ???
+        wave_a (np.ndarray):
```
Comment (JulianSKelly) [inline fpgalib/dac.py]: Blank arg

File: fpgalib/test/test_dac.py [hunk 2]
```diff
@@ -5,6 +5,21 @@
 import fpgalib.dac as dac
 
 
+def test_waves2sram():
```
Comment (JulianSKelly) [inline fpgalib/test/test_dac.py]: Name should be test_dacify

File: fpgalib/dac.py [hunk 3]
```diff
@@ -85,41 +85,43 @@
...
 # TODO: Integrate this with DAC class
 # TODO: Have dac_calibration_server not DACify SRAM data
-def dacify(wave_a, wave_b, trigger_idx=(16, 17)):
-    """Construct SRAM sequence for a list of waveforms.
+
+
+SRAM_MAX = 0x1FFF  # 8190 = 2**13 - 1
+
+
+def dacify(wave_a, wave_b, trigger_idx=None):
```
Comment (JulianSKelly) [inline fpgalib/dac.py]: Could you change the default triggers in fpgatester to compensate for default value removal?

File: fpgalib/dac.py [hunk 4]
```diff
@@ -85,41 +85,43 @@
...
-    # Chop off everything except lowest 14 bits.
-    trunc_a = [y & 0x3FFF for y in data_a]
-    trunc_b = [y & 0x3FFF for y in data_b]
-    dac_a_dat = trunc_a
-    # Shift DAC B data by 14 bits.
-    dac_b_dat = [y << 14 for y in trunc_b]
+
+    # Multiply wave by full scale. Then truncate to lowest 14 bits. Then for dac
+    # B shift left by 14 bits.
+    dac_a_dat = [0x3FFF & (long(np.floor(SRAM_MAX*y))) for y in wave_a]
```
Comment (JulianSKelly) [inline fpgalib/dac.py]: Why do we have to truncate when we verified it doesn't exceed +/-1? Are you truncating the bits below 14 bit resolution? Should the comment say truncate to largest 14 bits?

**Comment by JulianSKelly** [comment]
Minor comments and LGTM. Please change fpgatester trigger idx default.

File: fpgalib/dac.py (hunk 4 above)
Comment (DanielSank) [inline fpgalib/dac.py]: The truncation might not be necessary, but it was there before and I don't want to remove it without thinking carefully. As for the comment, it's definitely keeping the lowest bits, not the largest.

**Comment by DanielSank** [comment]
@JulianSKelly see latest commit

**Pull request merged**